msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java

```diff
@@ -82,7 +82,7 @@ public List<ICacheRecord> filter(@NonNull final List<ICacheRecord> records) {
             for (final ICacheRecord cacheRecord : records) {
                 final String acctHomeAccountId = cacheRecord.getAccount().getHomeAccountId();
                 final String acctLocalAccountId = cacheRecord.getAccount().getLocalAccountId();
-                if (acctLocalAccountId != null && acctHomeAccountId.contains(acctLocalAccountId)) {
```

Comment:
Why removing this null check? Isn't this going to actually cause NPE now?

Reply:
This was actually a band-aid fix of the same issue.

account local id is not supposed to be null, but a null check was added because we were getting null.

see melissa's suggestion in this PR
https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1933

Reply:
I looked into the code, the `getLocalAccountId` method on AccountRecord is not marked NonNull - so that means it is Nullable and that means that caller should perform the null check before using the value

Reply:
@shahzaibj the class AccountRecord was written in 2018 - that was before we introduced Nullable/NonNull into the code.

I'm not sure if we can conclude that every field in that class is nullable. (as in, if it's null, and assume we don't use the value, would there be anything break down the line)

The PR that introduced this null check was https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1847, which had the same symptom as https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1933.  We never had a null check here before then.